File: java/src/main/java/org/urlshortener/core/user/UserService.java
package org.urlshortener.core.user;

import lombok.NonNull;
import org.urlshortener.core.dtos.UserDto;

public interface UserService {
    UserDto findByName(@NonNull String name);

    UserDto findById(long id);

    void delete(long id);

    UserDto create(@NonNull UserDto userDto);

    UserDto authenticate(@NonNull String username, @NonNull String password);
}

File: kotlin/src/main/kotlin/com/urlshortener/core/services/user/UserService.kt
package com.urlshortener.core.services.user

import com.urlshortener.core.dtos.UserDto

interface UserService {
    fun findByName(name: String): UserDto
    fun findById(id: Long): UserDto?
    fun delete(id: Long)
    fun create(userDto: UserDto): UserDto
    fun authenticate(userDto: UserDto, password: String): Boolean
}